models/result.go

```diff
@@ -135,6 +136,24 @@ func (r *Result) HandleFormSubmit(details EventDetails) error {
 	return db.Save(r).Error
 }
 
+// HandleArbitraryEvent updates a Result with an arbitrary event (e.g Word document opened, secondary link clicked)
+func (r *Result) HandleArbitraryEvent(details EventDetails) error {
+
+	EventTitle := details.Payload.Get("title")
+
+	if EventTitle == "" {
```

Comment:
I think there might be a better place to put this. Maybe short-circuit this back in the handler?

🤔 I'll need to think about this a little more.